mlflow/gateway/providers/base.py

```diff
@@ -19,13 +19,22 @@ def __init__(self, config: RouteConfig):
         self.config = config
 
     async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
-        raise NotImplementedError
+        raise HTTPException(
+            status_code=404,
+            detail=f"The chat route is not available for {self.NAME} models.",
+        )
```

Comment:
tbh I think 501 status makes more sense, but not sure if this will break any user code.

Reply:
I agree. 501 (not implemented) makes more sense. Let's change it.